### ISSUE TITLE
Observability | Prometheus config

### ISSUE THREAD

**Issue by pataruco** [MHRA/products#524]
## User want

As a technical user 
I want set Prometheus to collect and digest doc index updater logs
So that I can enable monitoring and alerting tools

## Acceptance criteria

- [ ] Prometheus should collect and digest logs from doc index updater
- [ ] [Docs should explain how to see these logs
](https://github.com/MHRA/products/blob/master/infrastructure/docs/logs.md)

## Data - Potential impact

**Size**

**Value**

**Effort**

### Exit Criteria met 
- [x] Backlog
- [x] Discovery
- [x] DUXD
- [x] Development 
- [ ] Quality Assurance 
- [ ] Release and Validate

**Comment by rjkerrison** [comment]
@sgrowe do we have written instructions for viewing the prometheus logs anywhere?

**Comment by sgrowe** [comment]
@rjkerrison Not as yet but if you go to this ["logs analytics workspace"](https://portal.azure.com/#@MHRA.onmicrosoft.com/resource/subscriptions/bec11470-1346-4cdd-af2e-ce1f360671a1/resourceGroups/mhra-products-development/providers/Microsoft.OperationalInsights/workspaces/mhra-products-cluster-analytics-vzlE/logs) and run this search you should be able to view them:

```
InsightsMetrics
| where Namespace == "prometheus"
```

**Comment by craiga** [comment]
### Findings from QA

TL;DR: I found the documentation intimidating and it assumes a lot of knowledge, but it works.

_Caveat: I'm a first-time user of many of these tools and haven't had much to do with infrastructure things until now._

General query: Who is the intended audience for this document? I assume it's for developers, but how much are they expected to know about the infrastructure? Is there context to this document I should know about?

I'd appreciate a sentence at the start of the document explaining that this document explains how to get Prometheus and container logs. It would also be good to have a similar sentence at the start of the Prometheus section explaining that there are two ways of viewing Prometheus logs.

The first sentence of prose here is "Prometheus is a pod automatically included with istio". I (sort of) know what those things are, but I can easily imagine another less experienced developer arriving here being intimidated and scared away. Should the nouns here be links to the different projects' web sites?

It would be nice if `update-kubernetes-config.sh` was explained or maybe renamed. The document just links to the script which I assume means that I run it, but from the name of the file I was a bit worried that running it was going to push some update to the dev Kubernetes config. Renaming that script or adding a note explaining what the script does would help.

Some guidance or just a link to a document explaining how to use the Prometheus dashboard would be appreciated. As a first-time user, it took me a while. The first time UX is rubbish.

Regarding "Open the analytics workspace which is in the same resource group as your cluster", am I supposed to know this? How? As far as I can tell, the resource group prefixed with "ap" is production and the group prefixed with "ad" is non-prod; can this be included here?

I could only get log entries to appear in the Azure Log Analytics thing in the development environment, but I assume this is expected at this time.

It would be nice to have some common scenarios as to how we'd use stern. For example, how can I get log entries with a particular correlation ID?

The "Viewing container logs in Azure" section should either be expanded or removed.

**Comment by rjkerrison** [comment]
Raised #644 with ACs for each of the above, except:

> Regarding "Open the analytics workspace which is in the same resource group as your cluster", am I supposed to know this? How? As far as I can tell, the resource group prefixed with "ap" is production and the group prefixed with "ad" is non-prod; can this be included here?

I don't know what _ap_ and _ad_ are? The _resource group_ I'd usually be concerned about is `mhraproductsdevelopment`, or otherwise `mhraproductsnonprod`. I think if I were concerned about seeing logs, I would know which resource group I care about, as they're named one-to-one with our environments.

**Comment by m-doughty** [comment]
**QA Findings**

In Azure Log Workspace, I had to do:

```
InsightsMetrics
| where Namespace == "container.azm.ms/prometheus"
```

The document didn't contain the `container.azm.ms` part.

Other than that, it looks great.
